dotCMS/src/main/java/com/dotcms/contenttype/business/DotAssetAPIImpl.java

```diff
@@ -0,0 +1,148 @@
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Default implementation
+ * @author jsanca
+ */
+public class DotAssetAPIImpl implements DotAssetAPI {
+
+    @Override
+    public Optional<ContentType> tryMatch(final File file,final  Host currentHost,final  User user) throws DotDataException, DotSecurityException {
+
+        return this.tryMatch(MimeTypeUtils.getMimeType(file), currentHost, user);
+    }
+
+    @Override
+    public Optional<ContentType> tryMatch(final String mimeType, final Host currentHost, final User user) throws DotSecurityException, DotDataException {
+
+        final List<ContentType> dotAssetContentTypes = APILocator.getContentTypeAPI(user).findByType(BaseContentType.DOTASSET);
+
+        if (UtilMethods.isSet(dotAssetContentTypes)) {
+
+            // Stores the content type indexed by mimetypes, on each index stores the subsets
+            // 0:Exact/on Site, 1:Exact/SYSTEM_HOST, 2:Partial Wildcard/on Site, 3:Partial Wildcard/SYSTEM_HOST,
+            // 4:Total Wildcard (or null)/on Site, 5:Total Wildcard (or null)/SYSTEM_HOST
+            final Map<String, ContentType>[] mimeTypeMappingArray = new Map [6];
+
+            dotAssetContentTypes.stream()
+                    .filter(contentType -> isSystemHostOrCurrentHost(contentType, currentHost)) // remove the ones that are not system host or current host
+                    .map(this::mapToContentTypeMimeType).forEach(contentTypeMimeType -> {
+
+                for (final String mimeTypeItem : contentTypeMimeType.mimeTypeFieldVariables) {
+
+                    if (ALL_MIME_TYPE.equals(mimeTypeItem)) {
+
+                        this.getMap(mimeTypeMappingArray, contentTypeMimeType.systemHost?5:4).put(mimeTypeItem, contentTypeMimeType.contentType);
+                    } else if (mimeTypeItem.endsWith(PARTIAL_MIME_TYPE)) {
+
+                        this.getMap(mimeTypeMappingArray, contentTypeMimeType.systemHost?3:2).put(mimeTypeItem, contentTypeMimeType.contentType);
+                    } else {
+
+                        this.getMap(mimeTypeMappingArray, contentTypeMimeType.systemHost?1:0).put(mimeTypeItem, contentTypeMimeType.contentType);
+                    }
+                }
+            });
+
+            return findDotAssetContentType (mimeType, mimeTypeMappingArray);
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<ContentType> findDotAssetContentType(final String mimeType,
+                                                          final Map<String, ContentType>[] mimeTypeMappingArray) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028307)